Sankey docs (#2090)
telerik/blazor-docs: components/sankey/elements/legend.md

---
title: Legend
page_title: Sankey Legend
description: Legend of the Sankey Diagram for Blazor.
slug: sankey-legend
tags: telerik,blazor,sankey,diagram,chart,legend
published: True
position: 11
---
@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

# Sankey Legend

> The Telerik Sankey Diagram for Blazor can show a visual guide with details about the nodes. The Sankey renders a legend by design. This article explores how to customize the Sankey legend.

Reviewer: No need to burden the intro with implementation details.

```suggestion
The Telerik Sankey Diagram for Blazor can show a legend, which is a visual guide with details about the nodes. This article explores how to customize the Sankey legend.
```

## Basic Customization

> To customize the legend in the Sankey chart, declare a `<SankeyLegend>` tag as a direct child of `<TelerikSankey>`. The `<SankeyLegend>` tag has the following parameters:

Reviewer 2: For SEO purposes, consider going with `Blazor Sankey chart` here. Note that this is the first instance of Sankey chart, everywhere else it's been Sankey diagram.

| Parameter | Type and Default&nbsp;Value | Description |
| --------- | ---- | ----------- |
| `Align` | `SankeyLegendAlign` enum | The alignment of the legend. |
| `Background` | `string`  | The background color of the legend. |
| `Height` | `double?`  | The height of the legend. |
| `OffsetX` | `double?`  | The X offset of the legend. The offset is relative to the current position of the legend. |
| `OffsetY` | `double?`  | The Y offset of the legend. The offset is relative to the current position of the legend. |
| `Orientation` | `SankeyLegendOrientation` enum <br/> (`SankeyLegendOrientation.Horizontal`)  | The orientation of the legend. |
| `Position` | `SankeyLegendPosition` enum <br/> (`SankeyLegendPosition.Bottom`)| The position of the legend. |

> | `Reverse` | `double?`  | Whether the legend items are reversed. |

Reviewer 2: ```suggestion
| `Reverse` | `double?`  | Indicates whether the order of the legend items is reversed. |
```

| `Spacing` | `double?`  | The spacing between the labels of the legend. |

> | `Visible` | `bool?` <br/> (`true`) | Whether the legend is visible. |

Reviewer 2: ```suggestion
| `Visible` | `bool?` <br/> (`true`) | Indicates whether the legend is visible. |
```

| `Width` | `double?`  | The width of the legend. Applies when `Orientation="SankeyLegendOrientation.Horizontal"`. |

## Nested Customization Tags

> The `<SankeyLegend>` tag exposes nested tags for further customization of the separate legend elements. The structure of the nested tags is `<SankeyLegend*Specifics*>`, where the specifics can be:

Reviewer 2: ```suggestion
The `<SankeyLegend>` tag exposes nested tags for further customization of the separate legend elements. The structure of the nested tags is `<SankeyLegend*Specifics*>` with the following specifics:
```

* [`Border`](/blazor-ui/api/telerik.blazor.components.sankeylegendborder)
* [`Item`](/blazor-ui/api/telerik.blazor.components.sankeylegenditem)

> * [`Labels`](/blazor-ui/api/telerik.blazor.components.sankeylegendlabels) - exposes additional nested options. The structure of the nested tags is `<SankeyLegendLabels*Specifics*>`, where the specifics can be:

Reviewer 2: ```suggestion
* [`Labels`](/blazor-ui/api/telerik.blazor.components.sankeylegendlabels)&mdash;Exposes additional nested options. The structure of the nested tags is `<SankeyLegendLabels*Specifics*>` with the following specifics:
```

    * [`Margin`](/blazor-ui/api/telerik.blazor.components.sankeylegendlabelsmargin)
    * [`Padding`](/blazor-ui/api/telerik.blazor.components.sankeylegendlabelspadding)
* [`Margin`](/blazor-ui/api/telerik.blazor.components.sankeylegendmargin)
* [`Padding`](/blazor-ui/api/telerik.blazor.components.sankeylegendpadding)

> * [`Title`](/blazor-ui/api/telerik.blazor.components.sankeylegendtitle) - exposes additional nested options. The structure of the nested tags is `<SankeyLegendTitle*Specifics*>`, where the specifics can be:

Reviewer 2: ```suggestion
* [`Title`](/blazor-ui/api/telerik.blazor.components.sankeylegendtitle)&mdash;Exposes additional nested options. The structure of the nested tags is `<SankeyLegendTitle*Specifics*>` with the following specifics:
```

        Data.Links.Add(new SankeyDataLink() { SourceId = 1, TargetId = 4, Value = 4 });
        Data.Links.Add(new SankeyDataLink() { SourceId = 1, TargetId = 7, Value = 8 });
        Data.Links.Add(new SankeyDataLink() { SourceId = 2, TargetId = 4, Value = 4 });
        Data.Links.Add(new SankeyDataLink() { SourceId = 2, TargetId = 5, Value = 24 });
        Data.Links.Add(new SankeyDataLink() { SourceId = 2, TargetId = 6, Value = 10 });
        Data.Links.Add(new SankeyDataLink() { SourceId = 2, TargetId = 7, Value = 2 });
        Data.Links.Add(new SankeyDataLink() { SourceId = 3, TargetId = 5, Value = 11 });
        Data.Links.Add(new SankeyDataLink() { SourceId = 3, TargetId = 6, Value = 28 });
        Data.Links.Add(new SankeyDataLink() { SourceId = 3, TargetId = 7, Value = 9 });
    }

    #endregion Data generation
}
````

## See Also

* [Live Demo: Sankey Diagram Configuration](https://demos.telerik.com/blazor-ui/sankey/configuration)
* [Sankey Links]({%slug sankey-links%})
* [Sankey Nodes]({%slug sankey-nodes%})
* [Sankey Labels]({%slug sankey-labels%})

> * [Sankey Tooltip]({%slug sankey-sankey-tooltip%})

Reviewer: ```suggestion
* [Sankey Tooltip]({%slug sankey-tooltip%})
```
